tests/integration/test_reads.py

```diff
@@ -906,7 +906,7 @@ def test_table_scan_override_with_small_types(catalog: Catalog) -> None:
     expected_schema = pa.schema(
         [
             pa.field("string", pa.string()),
-            pa.field("string-to-binary", pa.binary()),
+            pa.field("string-to-binary", pa.large_binary()),
```

Comment:
@Fokko is this right? type promotion for string->binary results in a large_binary type
https://github.com/apache/iceberg-python/blob/7a56ddb680313057526a704a6105aea175120c34/pyiceberg/io/pyarrow.py#L687-L688

i found these 3 places where `_ConvertToArrowSchema` converts to large type by default
* [list](https://github.com/apache/iceberg-python/blob/7a56ddb680313057526a704a6105aea175120c34/pyiceberg/io/pyarrow.py#L629) 
* [string](https://github.com/apache/iceberg-python/blob/7a56ddb680313057526a704a6105aea175120c34/pyiceberg/io/pyarrow.py#L679)
* [binary](https://github.com/apache/iceberg-python/blob/7a56ddb680313057526a704a6105aea175120c34/pyiceberg/io/pyarrow.py#L688), as seen above